src/service.ts

```diff
@@ -400,24 +410,46 @@ export class GrpcService extends Service {
    * @param {object} reqOpts - The request options.
    * @param {function=} callback - The callback function.
    */
-  // @ts-ignore
-  request(protoOpts, reqOpts, callback) {
+  request(reqOpts: DecorateRequestOptions): Promise<r.Response>;
+  request(reqOpts: DecorateRequestOptions, callback: BodyResponseCallback):
+      void;
+  request(reqOpts: DecorateRequestOptions, callback?: BodyResponseCallback):
+      void|Promise<r.Response>;
+  request(
+      protoOpts: ProtoOpts, reqOpts: DecorateRequestOptions,
+      callback: ServiceRequestCallback): Abortable|void;
+  request(
+      pOpts: ProtoOpts|DecorateRequestOptions,
+      rOpts?: DecorateRequestOptions|BodyResponseCallback,
+      callback?: ServiceRequestCallback): Abortable|void|Promise<r.Response> {
```

Comment:
IMO, this really waters down the types we are providing to our users. If the method signatures don't actually map, then we really have no business claiming that `GrpcService` is a subtype of `Service`.

What functionality do we really need to inherit from `Service` anyway?

Reply:
That's a really good question, and one I asked myself as I went romping through the type signatures to craft this man-bear-pig of an API. Obviously any change here would be super breaking, and isn't really what I was looking to do with these PRs.  To be honest, this entire *library* needs to be rationalized as compared to gax. 

For now - lets pinch our noses, and get the types worked out so we can transition logging, spanner, and the other libs that use this.  Thoughts?